### ISSUE TITLE
[SOLVED] Start myExperiment

### ISSUE THREAD

**Issue by simone24** [myExperiment/myExperiment#15]
Hi,
sorry but I have two more problems on running myExperiment: 

1. I can't start myExperiment outside myExperiment folder (In my case /opt/myExperiment)
2. When I close the ssh session, myExperiment stops even if I use & in the end of the command

**_First issue_**
I noticed that I need to be in myExperiment folder to launch _script/server_ and run myExperiment.
So I wrote the file **start_myexp.sh** in **/etc/init.d**

_#!/bin/bash
cd /opt/myExperiment
script/server &_

I've tried to run this file from **/etc/init.d** but it returns the following message: 
_/opt/myExperiment/config/preinitializer.rb:8:in `rescue in <top (required)>': Could not load the bundler gem. Install it with `gem install bundler`. (RuntimeError)
        from /opt/myExperiment/config/preinitializer.rb:4:in `<top (required)>'
        from /opt/myExperiment/config/boot.rb:28:in `load'
        from /opt/myExperiment/config/boot.rb:28:in `preinitialize'
        from /opt/myExperiment/config/boot.rb:10:in `boot!'
        from /opt/myExperiment/config/boot.rb:128:in `<top (required)>'
        from <internal:lib/rubygems/custom_require>:29:in `require'
        from <internal:lib/rubygems/custom_require>:29:in `require'
        from /opt/myExperiment/script/server:2:in `<main>'_

But if I run **/etc/init.d/start_myexp.sh** from **/opt/myExperiment** it works.
How can I start myExperiment from outside /opt/myExperiment folder?

**_Second issue_**

When I run the command _script/server_ I put & in the end so I can close the terminal without stopping the service, but it doesn't work.
I also tried to use **-d** options (script/server -d) but it only returns 
_=> Booting WEBrick
=> Rails 2.3.18 application starting on http://0.0.0.0:3000_
After that it shows me the prompt but, when I go in the web browser, I see an empty page.

I've tried a lot of solutions and trick before writing here, but with bad results.

Thanks
Simone

**Comment by fbacall** [comment]
1. Try putting `source ~/.rvm/scripts/rvm` at the top of your script.
2. You need to use something like `nohup` or `screen` to keep it running after your SSH session closes. ```nohup ruby script/server &```

Are you intending to run myExperiment as a service, or just for development purposes?

We have an installation script that might be helpful if you are doing the former. It is mentioned at the bottom of this page: http://wiki.myexperiment.org/index.php/Developer:UbuntuInstallation#Installation_script_for_Ubuntu.2FDebian.2FLinux_Mint

although admittedly I haven't tried it recently.

**Comment by simone24** [comment]
With your help I modified the script and now I can run it from /etc/init.d and I can close the ssh session without stopping the service.
I also made
_update-rc.d script.sh defaults_ 
to run the script at boot, but it doesn't work.

In the meanwhile I created a fresh ubuntu machine and, after setting the file _setting.bash_ I run _install.bash_ but it returns

_Installing and configuring RVM
Install and configuring Ruby version 1.8.7
myExperiment-installers/ubuntu/install.bash: line 29: /usr/local/rvm/scripts/rvm: No such file or directory
Could not source /usr/local/rvm/scripts/rvm to add RVM directory to path. Aborting ..._

Thanks
